mypy/checker.py

```diff
@@ -287,6 +287,26 @@ class PartialTypeScope(NamedTuple):
     is_local: bool
 
 
+class InstanceDeprecatedVisitor(TypeTraverserVisitor):
+    """Visitor that recursively checks for deprecations in nested instances."""
+
+    def __init__(self, typechecker: TypeChecker, context: Context) -> None:
+        self.typechecker = typechecker
+        self.context = context
+        self.seen_aliases: set[TypeAliasType] = set()
+
+    def visit_instance(self, t: Instance) -> None:
+        super().visit_instance(t)
+        self.typechecker.check_deprecated(t.type, self.context)
+
+    def visit_type_alias_type(self, t: TypeAliasType) -> None:
+        super().visit_type_alias_type(t)
+        if t not in self.seen_aliases:
+            self.seen_aliases.add(t)
+            if ((alias := t.alias) is not None) and ((target := alias.target) is not None):
+                target.accept(self)
```

Comment:
You should visit `.args`, otherwise things like this will not be caught:
```python
type Pairs[T] = list[tuple[T, T]]

x: Pairs[Deprecated]
```

Reply:
I removed the method.